include/flatbuffers/flatbuffers.h

```diff
@@ -393,6 +393,84 @@ template<typename T> static inline size_t VectorLength(const Vector<T> *v) {
   return v ? v->size() : 0;
 }
 
+// This is used as a helper type for accessing arrays.
+template<typename T, int length> class Array {
+ public:
+  typedef VectorIterator<T, typename IndirectHelper<T>::mutable_return_type>
+      iterator;
+  typedef VectorIterator<T, typename IndirectHelper<T>::return_type>
+      const_iterator;
+  typedef VectorReverseIterator<iterator> reverse_iterator;
+  typedef VectorReverseIterator<const_iterator> const_reverse_iterator;
+
+  uoffset_t size() const { return length; }
+
+  typedef typename IndirectHelper<T>::return_type return_type;
+  typedef typename IndirectHelper<T>::mutable_return_type mutable_return_type;
+
+  return_type Get(uoffset_t i) const {
+    FLATBUFFERS_ASSERT(i < size());
+    return IndirectHelper<T>::Read(Data(), i);
+  }
+
+  return_type operator[](uoffset_t i) const { return Get(i); }
+
+  iterator begin() { return iterator(Data(), 0); }
+  const_iterator begin() const { return const_iterator(Data(), 0); }
+
+  iterator end() { return iterator(Data(), size()); }
+  const_iterator end() const { return const_iterator(Data(), size()); }
+
+  reverse_iterator rbegin() { return reverse_iterator(end()); }
+  const_reverse_iterator rbegin() const { return const_reverse_iterator(end()); }
+
+  reverse_iterator rend() { return reverse_iterator(end()); }
+  const_reverse_iterator rend() const { return const_reverse_iterator(end()); }
+
+  const_iterator cbegin() const { return begin(); }
+
+  const_iterator cend() const { return end(); }
+
+  const_reverse_iterator crbegin() const { return rbegin(); }
+
+  const_reverse_iterator crend() const { return rend(); }
+
+  // Change elements if you have a non-const pointer to this object.
+  void Mutate(uoffset_t i, const T &val) {
+    FLATBUFFERS_ASSERT(i < size());
+    WriteScalar(data() + i, val);
+  }
+
+  // Get a mutable pointer to tables/strings inside this array.
+  mutable_return_type GetMutableObject(uoffset_t i) const {
+    FLATBUFFERS_ASSERT(i < size());
+    return const_cast<mutable_return_type>(IndirectHelper<T>::Read(Data(), i));
+  }
+
+  // The raw data in little endian format. Use with care.
```

Comment:
What does `little endian` mean?

Reply:
Little endian is the most common storage order of integers with the least significant byte first.
Big endian has the opposite order. There are also other exotic orders found on old machines and in cryptography. The name originates from the childrens book Gulliver's travels where he visits the two neighbouring islands where one population eats their eggs with the big end up, and the other population eats their eggs with the little end up.

Reply:
I read Gulliver :)
Lets I have BE machine and declare an array:
```cpp
int arr[3]={1,2,3};
auto pu8 = reinterpret_cast<const uint8_t*>(&arr[0]);
```
By nature, the `pu8` is neither BE nor LE.
It points to memory with BE integers (not to LE).

Reply:
I was wrong, but have a new question.
Scalar is always represented in little-endian format. Array of scalars also `LE`.
```cpp
// The raw data in little endian format. Use with care.
uint8_t *Data() { return reinterpret_cast<uint8_t *>(&length_ + 1); }
const T *data() const { return reinterpret_cast<const T *>(Data()); }
```
How to use `data()` on `BE` machine?

Reply:
I'll hazard a guess. This is a low-level pointer access method that is not endian safe. Hence the "use with care" comment. In flatcc for C, the method is called _get_ptr() where the safe method is _get(i) returning the field value.

For 8-bit elements it doesn't matter, but uint16_t would be unsafe. I have seen users getting into such problems on big endian hardware.

Reply:
Right you have to use `Get(i)` and `Mutate(i, val)`.